### PR TITLE
pipeline: add renderRBAC func

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

part of #493

### CODE REVIEW

File: pkg/fleet-manager/pipeline/render/rbac.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+// RBACConfig contains the configuration data required for the RBAC template.
+// Both PipelineName and PipelineNamespace are required.
+type RBACConfig struct {
+	PipelineName      string // Name of the pipeline.
+	PipelineNamespace string // Kubernetes namespace where the pipeline is deployed.
+}
+
+// ServiceAccountName generates the service account name using the pipeline name and namespace.
+func (rbac RBACConfig) ServiceAccountName() string {
+	return ServiceAccountNamePrefix + rbac.PipelineName + "-" + rbac.PipelineNamespace
```
Comment (hzxuzhonghu) [inline pkg/fleet-manager/pipeline/render/rbac.go:42]: seems we donot need the `ServiceAccountNamePrefix`

Comment (hzxuzhonghu) [reply pkg/fleet-manager/pipeline/render/rbac.go:42]: And neither pipelinenamespace

File: pkg/fleet-manager/pipeline/render/rbac.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"fmt"
+	"io/fs"
+)
+
+const (
+	ServiceAccountNamePrefix     = "kurator-pipeline-robot-"
+	RoleBindingNameSuffix        = "-binding"
+	ClusterRoleBindingNameSuffix = "-clusterbinding"
```
Comment (hzxuzhonghu) [inline pkg/fleet-manager/pipeline/render/rbac.go:27]: not such naming convention, `clusterrolebinding` or `rolebinding` is commonly used

File: pkg/fleet-manager/pipeline/render/rbac.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+// ServiceAccountName generates the service account name using the pipeline name and namespace.
+func (rbac RBACConfig) ServiceAccountName() string {
+	return rbac.PipelineName
+}
+
+// RoleBindingName generates the role binding name using the service account name.
+func (rbac RBACConfig) RoleBindingName() string {
+	return rbac.ServiceAccountName() + RoleBindingNameSuffix
```
Comment (hzxuzhonghu) [inline pkg/fleet-manager/pipeline/render/rbac.go:46]: nit: no need to add this suffix

Comment (Xieql) [reply pkg/fleet-manager/pipeline/render/rbac.go:46]: ok

File: pkg/fleet-manager/pipeline/render/rbac.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+// RoleBindingName generates the role binding name using the service account name.
+func (rbac RBACConfig) RoleBindingName() string {
+	return rbac.ServiceAccountName()
+}
+
+// ClusterRoleBindingName generates the cluster role binding name using the service account name.
+func (rbac RBACConfig) ClusterRoleBindingName() string {
+	return rbac.ServiceAccountName()
```
Comment (hzxuzhonghu) [inline pkg/fleet-manager/pipeline/render/rbac.go:49]: Sorry just note this is cluster scope resource, if we create multiple clusterrolebinding, they conflict with same name.

Option one is to make it with different name, alternatively add different sa within the same clusterrolebinding.

File: pkg/fleet-manager/pipeline/render/rbac_test.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"kurator.dev/kurator/pkg/fleet-manager/manifests"
+)
+
+var manifestFS = manifests.BuiltinOrDir("manifests")
```
[the remark this answers is not in the document]
Comment (Xieql) [reply pkg/fleet-manager/pipeline/render/rbac_test.go:28]: this is copy from https://github.com/kurator-dev/kurator/blob/90645c2e65529bd218be9d95ebc4e87c0c59c85e/pkg/fleet-manager/plugin/render.go#L87-L119

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**